pkg/build/metrics/prometheus/metrics.go

```diff
@@ -77,40 +96,58 @@ func (bc *buildCollector) Collect(ch chan<- prometheus.Metric) {
 
 	// since we do not collect terminal build metrics on a per build basis, collectBuild will return counts
 	// to be added to the total amount posted to prometheus
-	var failed, error, cancelled, complete int
+	var error, cancelled, complete int
+	reasons := map[string]int{}
 	for _, b := range result {
-		f, e, cc, cp := bc.collectBuild(ch, b)
-		failed = failed + f
+		e, cc, cp, r := bc.collectBuild(ch, b)
+		for key, value := range r {
+			count := reasons[key]
+			count = count + value
```

Comment:
count:=reasons[value]+value ?

Reply:
or better reasons[key]=reasons[key]+value?

Reply:
yep ... iterated a bit on this piece and didn't finish clean up .. thx